lib/sentry/client.ex

```diff
@@ -218,6 +218,10 @@ defmodule Sentry.Client do
       e ->
         {:error, e}
     end
+  catch
+    :exit, data -> {:error, {:exit, data, __STACKTRACE__}}
+    :throw, data -> {:error, {:throw, data, __STACKTRACE__}}
+    :error, data -> {:error, {:error, data, __STACKTRACE__}}
```

Comment:
It'd be easier to log directly instead of reassembling the error in `maybe_log_result/1`.

Reply:
I'd be happy to see what that would look like!

Reply:
I've added another commit with the changes I had in mind. I hope you like them :)

Reply:
@mitchellhenke ping :)